src/Sylius/Bundle/CoreBundle/Resources/config/doctrine/model/Channel.orm.xml

```diff
@@ -17,6 +17,12 @@
                                       http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
 
     <mapped-superclass name="Sylius\Component\Core\Model\Channel" table="sylius_channel">
+        <many-to-one field="defaultLocale" target-entity="Sylius\Component\Locale\Model\LocaleInterface">
+            <join-column name="default_locale_id" referenced-column-name="id" nullable="true" on-delete="SET NULL"/>
+        </many-to-one>
+        <many-to-one field="defaultCurrency" target-entity="Sylius\Component\Currency\Model\CurrencyInterface">
+            <join-column name="default_currency_id" referenced-column-name="id" nullable="true" on-delete="SET NULL"/>
+        </many-to-one>
```

Comment:
I made the fields nullable, although I'm not sure this is the way to go. How currencies/locales are set up right now doesn't allow me to make these not nullable. 

But perhaps we can fallback on the first locale/currency from the available ones that are set for the channel when the defaults are null. And if no locales/currencies are added we can fallback on the applications default locale/currency.